torch/csrc/autograd/record_function.cpp

```diff
@@ -8,20 +8,22 @@ namespace torch { namespace autograd { namespace profiler {
 namespace {
 std::vector<RecordFunctionCallback> start_callbacks;
 std::vector<RecordFunctionCallback> end_callbacks;
+std::vector<bool> is_callback_sampled;
+size_t num_sampled_callbacks = 0;
 size_t callback_needs_inputs = 0;
 thread_local RecordFunction* thread_local_func_ = nullptr;
 
-bool is_sampled_callbacks = false;
+bool sampling_prop_set = false;
 double sampling_prob = 1.0;
 constexpr double kEps = 1e-10;
 }
 
 void setSamplingProbability(double prob) {
   if (std::abs(prob - 1.0) < kEps) {
```

Comment:
If someone passes 1.0 it will be exactly equal. Why do we need this episilon stuff? It strikes me as cargo-culting not using floating point equality. But this is a comparison against 1.0, which is perfectly representable and users will just call setSamplingProbability(1.0) if they want it always on.

Reply:
I was concerned about potential conversions (float/double), arithmetics and other ways an internal representation of 1.0 might potentially change to something not equal to 1.0 exactly, but if this is not an issue, I can simplify this to == with 1.0